modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc

```diff
@@ -0,0 +1,275 @@
+                        DISTINCT AcquisitionProtocolID))";
+                }
+                $isFirst = false;
+            }
+            $acqpif       .= ",'new','')";
+            $newQueryCase .= ")";
+        } else {
+            $acqpid        = '';
+            $acqpif       = "'new'";
+            $newQueryCase ='';
+        }
+
+        $NewDataSubquery = "
+            CASE 
+                COALESCE(Max(fqc.QCLastChangeTime), 'new')
+                WHEN 'new' THEN {$acqpif}
+                WHEN ''    THEN {$acqpif}
+                WHEN NULL  THEN {$acqpif}
+                ELSE ''
+            END";
+
+
+        // =====================================================
+        // Create Pending Fail and Pending New subquery
+        // =====================================================
+
+        foreach ($scan_id_types as $id => $type) {
+            $pass[$id]          = $type . 'pass';
+            $qc[$id]            = $type . 'QC';
+            $coalesce_desc[$id] = $pass[$id] . '.' . $qc[$id];
+            $case_desc[$id]     = "
+                CASE
+                    COALESCE($coalesce_desc[$id], '')
+                    WHEN '' THEN ''
+                    WHEN 1 THEN 'Passed'
+                    WHEN 2 THEN 'Failed'
+                END";
+        }
+
+        $PendingFailSubquery = "
+            CASE s.MRIQCStatus
+                WHEN 'Fail' THEN
+                    IF(s.MRIQCPending='Y', 'Pending Fail', 'Fail')
+                WHEN 'Pass' THEN
+                    IF(s.MRIQCPending='Y', 'Pending Pass', 'Pass') 
+                ELSE s.MRIQCStatus
+            END 
+            ";
+
+        $PendingNewquery = "
+            CASE 
+                WHEN s.MRIQCPending='Y' THEN 'P'
+                WHEN MAX(fqc.QCFirstChangeTime) IS NULL $newQueryCase THEN  'N'      
+            END";
+
+
+        // ====================================================
+        // Create left join to include QC status for specific
+        // scans in the data table
+        // ====================================================
+
+        $left_joins = "";
+        foreach ($case_desc as $key => $value) {
+            $left_joins .= "
+            LEFT JOIN (
+              SELECT files.SessionID, 
+                MIN(files_qcstatus.QCStatus+0) as $qc[$key] 
+              FROM files 
+              JOIN files_qcstatus USING (FileID) 
+              WHERE files.AcquisitionProtocolID= $key 
+                AND files_qcstatus.QCStatus IN (1, 2) 
+              GROUP BY files.SessionID) $pass[$key] 
+                ON ($pass[$key].SessionID=f.SessionID
+            ) ";
+        }
+
+        
+        // ===================================================
+        // Determine which data should be returned depending
+        // on user permission
+        // ===================================================
+
+        if (!$user->hasPermission('imaging_browser_view_allsites')) {
```

Comment:
I think these should be done with filters on the base provisioner similar to the UserSiteMatch filter, not by concatenating changes to the SQL

Reply:
haha. That was another one I wanted to ask you but totally forgot about it. I guess my suspicions that it was not right were correct ;).

Reply:
should be done in the last commit. @driusan was it what you expected?

Note: that breaks the candidate list as it expects that the new SiteFilters is called in the data provisioner of the candidate list. I have a feeling I missed something cause it is weird to have to change all the modules every time we have a new permission to use for filters.

Reply:
I don't see any changes to the candidate list in the last commit, so I'm assuming you didn't push it.. but if I can guess what the problems are:

1. You should just overload the `getDataProvisionerWithFilters` instead of doing it in the base class
2. It might make sense to have the base class refactored to have a function which returns a list of types of per-site permissions instead of a boolean, but that might require quite a big refactoring

Reply:
did suggestion 1. in the last commit